app/controllers/spree/affirm_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Spree
+  class AffirmController < Spree::StoreController
+    protect_from_forgery except: [:confirm]
+
+    def confirm
+      order = current_order || raise(ActiveRecord::RecordNotFound)
```

Comment:
Yes, in that we don't have an order in the session. But how would a user ever be redirected here if they didn't have an order in the session anyway? Odds are we would want some more helpful information here to assist the user in recovering _or_ we fail hard as a precondition that would notify us that something is wrong.